pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -376,33 +376,30 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 			return reconcile.Result{}, err
 		}
 
-		if authenticationCR != nil && authenticationCR.Status.State != operatorv1.TigeraStatusReady {
-			r.status.SetDegraded(operatorv1.ResourceNotReady,
-				fmt.Sprintf("Authentication is not ready authenticationCR status: %s", authenticationCR.Status.State),
-				nil, reqLogger)
-			return reconcile.Result{}, nil
-		} else if authenticationCR != nil && !utils.IsDexDisabled(authenticationCR) {
-			// Do not include DEX TLS Secret Name if authentication CR does not have type Dex
-			secret := render.DexTLSSecretName
-			certificate, err := certificateManager.GetCertificate(r.client, secret, common.OperatorNamespace())
+		if authenticationCR != nil && authenticationCR.Status.State == operatorv1.TigeraStatusReady {
+			if !utils.IsDexDisabled(authenticationCR) {
```

Comment:
Would you mind adding a `utils.IsDexEnabled` function and remove this double negative (negating isDisabled)?

Reply:
Added a commit. Much better, I can't stand it either ;)